### PR TITLE
Response invalid fix/Limit returned movies/Cancel intent

### PR DESCRIPTION
**Pull request by alexc-MSFT**
Rebuild new array of returned movies with minimum required movie data to
fix response invalid on Echo devices (due to response size limitation).

Also used node-couchpotato to limit the returned number of movies to 5 as opposed to slicing the array. 

Removed duplicate cancel intent and added a cancel intent response.

Made a whole bunch of commits now as I made a ton of changes locally and it got messy!

The sooner this pull request is merged I can add more functionality that i've written - i'll create a new branch this time!

### CODE REVIEW

File: lib/handlers.js [hunk 1]
```diff
@@ -7,11 +7,13 @@ var WELCOME_DESCRIPTION = 'This skill allows you to manage your Couch Potato mov
...
     'to it. Try asking "is The Godfather on the list?". If it\'s not and you want to add it, try',
     'saying "add The Godfather".'].join(' ');
+var CANCEL_RESPONSE = 'Exiting Couch Potato';
 
 var config = require('dotenv').config();
 var cp = new CouchPotato({
   url: config.CP_URL,
-  apikey: config.CP_API_KEY
+  apikey: config.CP_API_KEY,
+  debug:true
```
Comment (josephschmitt) [inline lib/handlers.js:16]: Do we want to leave debug on? What does it do?

File: lib/utils.js [hunk 2]
```diff
@@ -24,12 +24,42 @@ function sendSearchResponse(movies, resp) {
 
   resp
     .say(['Add', movies[0].original_title, '(' + movies[0].year + ')', 'to your list?'].join(' '))
-    .session('promptData', buildPrompt(movies.slice(0, 5)))
+    .session('promptData', buildPrompt(movies))
     .shouldEndSession(false)
     .send();
+
+function formatSearchResults(movies)
+{
```
Comment (josephschmitt) [inline lib/utils.js]: Sorry to be picky, but do you mind updating all these open curlies to be on the same line as the function? Just want the formatting to internally consistent in this file.

**Review by josephschmitt — COMMENTED** [review]
Nitpicky styling stuff otherwise ship it!

File: lib/handlers.js (hunk 1 above)
Comment (alexc-MSFT) [reply lib/handlers.js:16]: Returns some more info in the response - might be worth it whilst continuing to dev - shouldnt effect people using it.

File: lib/utils.js (hunk 2 above)
Comment (alexc-MSFT) [reply lib/utils.js]: Done after a ton of commits - I think

File: lib/handlers.js [hunk 3]
```diff
@@ -52,10 +54,16 @@ function handleFindMovieIntent(req, resp) {
...
+ 
+ cp.movie.search(movieName,5).then(function (movies) {
+   
+   movies = utils.formatSearchResults(movies);
+   utils.sendSearchResponse(movies, movieName, resp);
+ })
+ 
+  //Async response
+  return false;
+}
```
Comment (hackworth) [inline lib/handlers.js:66]: Duplicate return false

File: lib/utils.js [hunk 4]
```diff
@@ -24,12 +24,38 @@ function sendSearchResponse(movies, resp) {
...
+  
+    if (movies != undefined) {
+
+      var newMovies = [];
+
+      for (var i = 0; i < movies.length; i++) {
+
+        newMovies.push({
+          original_title: movies[i].original_title,
+          inLibrary: isInLibrary(movies[i].in_library),
```
Comment (hackworth) [inline lib/utils.js]: in_library not inLibrary

File: lib/handlers.js [hunk 5]
```diff
@@ -52,10 +54,16 @@ function handleFindMovieIntent(req, resp) {
...
   var movieName = req.slot('movieName');
-
-  cp.movie.search(movieName).then(function (movies) {
-    utils.sendSearchResponse(movies, resp);
-  });
+ 
+ cp.movie.search(movieName,5).then(function (movies) {
+   
+   movies = utils.formatSearchResults(movies);
+   utils.sendSearchResponse(movies, movieName, resp);
```
Comment (hackworth) [inline lib/handlers.js:60]: Sending movieName?

**Review by hackworth — CHANGES_REQUESTED** [review]

File: lib/utils.js (hunk 4 above)
Comment (alexc-MSFT) [reply lib/utils.js]: I added a new function to check if the movie is already in your library (different json is defined) but I didn't commit that function - removed the reference to it.

File: lib/handlers.js (hunk 5 above)
Comment (alexc-MSFT) [reply lib/handlers.js:60]: That's because in sendSearchResponse it looks for movieName which was not being passed in

File: lib/handlers.js (hunk 3 above)
Comment (alexc-MSFT) [reply lib/handlers.js:66]: Fixed - ta

**Comment by alexc-MSFT** [comment]
Clearly I need to get better at Git (sorry i'm new to it) - made a ton of commits but I think I got there in the end!

**Comment by josephschmitt** [comment]
@sharepointalex @Hackworth this is awesome, you guys rock.

**Pull request merged**